connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/InboundConnectorBeanDefinitionProcessor.java

```diff
@@ -0,0 +1,129 @@
+    for (String beanName : handlerBeans) {
+      BeanDefinition beanDefinition = registry.getBeanDefinition(beanName);
+      InboundConnector inboundConnectorAnnotation =
+          listableBeanFactory.findAnnotationOnBean(beanName, InboundConnector.class);
+      if (inboundConnectorAnnotation != null) {
+        InboundConnectorProperties properties =
+            getProperties(inboundConnectorAnnotation, beanDefinition, beanName);
+        preparedProperties.add(properties);
+      }
+    }
+  }
+
+  private InboundConnectorProperties getProperties(
+      InboundConnector inboundConnector, BeanDefinition beanDefinition, String beanName) {
+    var scope = beanDefinition.getScope();
+    if (!SCOPE_PROTOTYPE.equals(scope)) {
+      throw new IllegalStateException(
+          "Only \""
+              + SCOPE_PROTOTYPE
+              + "\" scope is supported for inbound connectors but found: \""
+              + scope
+              + "\" for bean: "
+              + beanDefinition);
+    } else {
+      InboundConnectorProperties properties =
+          getInboundConnectorProperties(inboundConnector, beanDefinition, beanName);
+      return properties;
+    }
+  }
+
+  private InboundConnectorProperties getInboundConnectorProperties(
+      InboundConnector inboundConnector, BeanDefinition beanDefinition, String beanName) {
+    var deduplicationProperties = Arrays.asList(inboundConnector.deduplicationProperties());
+    Class<? extends InboundConnectorExecutable> inboundConnectorClass;
+    try {
+      inboundConnectorClass =
+          (Class<? extends InboundConnectorExecutable>)
+              Class.forName(beanDefinition.getBeanClassName());
+    } catch (ClassNotFoundException e) {
+      throw new RuntimeException(
+          "Error while loading class for bean with class name " + beanDefinition.getBeanClassName(),
+          e);
+    }
+    return new InboundConnectorProperties(
+        beanName,
+        inboundConnector.name(),
+        inboundConnector.type(),
+        inboundConnectorClass,
+        deduplicationProperties);
+  }
+
+  @Override
+  public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory)
+      throws BeansException {
+    InboundConnectorFactory inboundConnectorFactory =
+        beanFactory.getBean(InboundConnectorFactory.class);
+    preparedProperties.stream()
+        .map(p -> fromProperties(p, beanFactory))
+        .forEach(inboundConnectorFactory::registerConfiguration);
+  }
+
+  private InboundConnectorConfiguration fromProperties(
+      InboundConnectorProperties properties, BeanFactory beanFactory) {
+    LOG.info("Configuring inbound connector {} of bean '{}'", properties, properties.beanName());
+    return new InboundConnectorConfiguration(
+        properties.name(),
+        properties.type(),
+        properties.clazz(),
+        () -> (InboundConnectorExecutable) beanFactory.getBean(properties.beanName()),
```

Comment:
```suggestion
        () -> beanFactory.getBean(properties.beanName(), properties.clazz()),
```
or (see previous comment)
```suggestion
        () -> beanFactory.getBean(properties.beanName(), InboundConnectorExecutable.class),
```